include/zephyr/drivers/battery.h

```diff
@@ -0,0 +1,470 @@
+ * See driver/battery/[your_driver].h/c for details on implementation and
+ * how the average is calculated.
+ */
+typedef int (*battery_get_avg_current)(const struct device *dev); /* in mA */
+typedef int (*battery_get_avg_voltage)(const struct device *dev); /* in mV */
+
+/**
+ * Return vendor-provided battery constants.
+ */
+typedef const struct battery_info *(*battery_get_info)(const struct device *dev);
+
+/**
+ * Get current battery parameters.
+ *
+ * Error conditions are reported via batt.flags.
+ *
+ * @param batt		Destination for battery data
+ */
+typedef void (*battery_get_params)(const struct device *dev, struct batt_params *batt);
+
+/**
+ * Modify battery parameters to match vendor charging profile.
+ *
+ * @param batt		Battery parameters to modify
+ */
+typedef void (*battery_override_params)(const struct device *dev, struct batt_params *batt);
+
+/**
+ * Check for presence of battery.
+ *
+ * @return Whether there is a battery attached or not, or if we can't tell.
+ */
+typedef enum battery_present (*battery_is_present)(const struct device *dev);
+
+/**
+ * Check for physical presence of battery.
+ *
+ * @return Whether there is a battery physically present, but possibly
+ * in a disconnected or cut off state, or if we can't tell;
+ */
+typedef enum battery_present (*battery_hw_present)(const struct device *dev);
+
+/**
+ * Check for battery initialization status.
+ *
+ * @return zero if not initialized.
+ */
+typedef int (*board_battery_initialized)(const struct device *dev);
+
+/**
+ * Read nominal voltage battery is designed to supply.
+ *
+ * @param voltage	Destination for voltage in mW
+ * @return non-zero if error.
+ */
+typedef int (*battery_design_voltage)(const struct device *dev, int *voltage);
+
+/**
+ * Read absolute state of charge.
+ *
+ * @param percent	Destination for charge in percent
+ * @return non-zero if error.
+ */
+typedef int (*battery_state_of_charge_abs)(const struct device *dev, int *percent);
+
+/**
+ * Read battery remaining capacity.
+ *
+ * @param capacity	Destination for capacity in mAh
+ * @return non-zero if error.
+ */
+typedef int (*battery_remaining_capacity)(const struct device *dev, int *capacity);
+
+/**
+ * Read battery full charge capacity.
+ *
+ * @param capacity	Destination for capacity in mAh
+ * @return non-zero if error.
+ */
+typedef int (*battery_full_charge_capacity)(const struct device *dev, int *capacity);
+
+/**
+ * Read the nominal capacity the battery is designed to supply when new.
+ *
+ * @param capacity	Destination for capacity in mAh
+ * @return non-zero if error.
+ */
+typedef int (*battery_design_capacity)(const struct device *dev, int *capacity);
+
+/**
+ * Read time in minutes left when discharging.
+ *
+ * @param capacity	Destination for remaining time in minutes.
+ * @return non-zero if error.
+ */
+typedef int (*battery_time_to_empty)(const struct device *dev, int *minutes);
+
+/**
+ * Read run time in minutes left when discharging.
```

Comment:
Explain in docblock difference between 'run_time_to_empty' and 'time _to_empty', both in this docblock & in the other.